### PR TITLE
monitoring: broadcasting funds alertmanager integration

### PR DESCRIPTION
**Pull request by kyriediculous**
This PR enables prometheus alertmanager for monitoring broadcasting funds according to the following rules:

- deposit below 0.2 ETH (page)
- deposit below 0.05 ETH (high) 
- reserve less than previous value (critical)

The current receiver is pagerduty for which a _service key_ is required. 

image

### CODE REVIEW

**Comment by kyriediculous** [comment]
TODO: update readme

File: monitoring/README.md [hunk 1]
```diff
@@ -46,5 +46,8 @@ $ sudo docker run --net=host livepeer/monitoring:latest --mode standalone --node
 
 ```
 
+**note** on OSX omit ``--net=host`  and use `docker.host.internal` instead of localhost for the `--nodes`.
```
Comment (yondonfu) [inline monitoring/README.md:49]: I had to use `host.docker.internal` to get this to work (also noted by the [docs](https://docs.docker.com/docker-for-mac/networking/)).

Comment (kyriediculous) [reply monitoring/README.md:49]: Oh sorry I turned this into docker.host.internal instead , my bad

File: monitoring/config-generator/src/generate.js [hunk 2]
```diff
@@ -164,362 +185,461 @@ function prometheusConfig (params) {
...
+    routes: [],
+  }
+
+  let receivers = []
+
+  // Inhibition rules allow to mute a set of alerts given that another alert is firing
+  let inhibit_rules = []
+  
+  inhibit_rules.push({
+    source_match: {
```
[the remark this answers is not in the document]
Comment (kyriediculous) [reply monitoring/config-generator/src/generate.js:569]: Yes, good point

File: monitoring/config-generator/src/generate.js [hunk 3]
```diff
@@ -69,93 +73,110 @@ function generate () {
...
       'node-exporter-port': {
-        describe: '[docker compose mode only] the port defined for node-exporter',
+        describe:
+          '[docker compose mode only] the port defined for node-exporter',
         default: 9100,
         type: 'number'
+      },
+      'pagerduty-service-key': {
+        descibe: 'pagerduty service key for alertmanager to send alerts',
+        default: 'a6c500da75e84d788e799bb00f6b08ae',
```
[the remark this answers is not in the document]
Comment (kyriediculous) [reply monitoring/config-generator/src/generate.js:86]: This is from the free trial account I made, forgot to remove. Should be an empty string indeed.

**Comment by yondonfu** [comment]
Tested this locally and the PagerDuty alerts seem to be working. After the requested changes are addressed and the README is updated, the PR will LGTM. I can then start testing this with a mainnet B cluster.

File: monitoring/README.md (hunk 1 above)
Comment (kyriediculous) [reply monitoring/README.md]: fixed in c15c362

File: monitoring/config-generator/src/generate.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (kyriediculous) [reply monitoring/config-generator/src/generate.js:570]: fixed in c15c362

File: monitoring/config-generator/src/generate.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (kyriediculous) [reply monitoring/config-generator/src/generate.js]: fixed in c15c362

File: monitoring/config-generator/src/generate.js [hunk 4]
```diff
@@ -164,362 +185,481 @@ function prometheusConfig (params) {
...
+    routes: [],
+  }
+
+  let receivers = []
+
+  // Inhibition rules allow to mute a set of alerts given that another alert is firing
+  let inhibit_rules = []
+  
+  inhibit_rules.push(
+    {
```
Comment (yondonfu) [inline monitoring/config-generator/src/generate.js:569]: Looks like this doesn't need to be indented.

Comment (kyriediculous) [reply monitoring/config-generator/src/generate.js:569]: The repo doesn't have a linter (we should do this in every JS repo and have a company standard on which settings to use). I'll run beautify for now.

Comment (kyriediculous) [reply monitoring/config-generator/src/generate.js:569]: There is a check if there is no service key provider: we return an empty object for the receiver.

Alertmanager will still process alerts and fire them but there will be no service to send them to I think. 

Haven't really tested out what happens when we omit it.

Comment (yondonfu) [reply monitoring/config-generator/src/generate.js:569]: Yeah that's a good idea. In this case, we can just be consistent with the indentation we're introducing with this set of changes and then a linter can be introduced separately.

**Comment by kyriediculous** [comment]
Added the pagerduty flag to the readme 88f2aa1

**Comment by kyriediculous** [comment]
Beautified the javascript file in ced6226

**Comment by kyriediculous** [comment]
rebased

**Pull request merged**